### ISSUE TITLE
(Java) //:tink fails to build without an Android toolchain present

### ISSUE THREAD

**Issue by notnarb** [google/tink#580]
**Background:**

I'm trying to import Tink  to my Bazel project to develop against some unreleased features in Master.

**Describe the bug:**

Building `//:tink` in java_src fails on my machine presumably because I don't have an Android toolchain configured.

**What was the expected behavior?**

`//:tink` I would assume should not require an Android SDK configured (`//:tink-android` on the other hand should).

**How can we reproduce the bug?**

(On a machine that does not have the Android SDK configured)

```bash
$ cd java_src
$ bazel build //:tink
...
ERROR: While resolving toolchains for target //src/main/java/com/google/crypto/tink:pem_key_type: no matching toolchains found for types @bazel_tools//tools/android:sdk_toolchain_type
ERROR: Analysis of target '//:tink' failed; build aborted: no matching toolchains found for types @bazel_tools//tools/android:sdk_toolchain_type
```

A dependency on android rules can be seen with `bazel query 'somepath(//:tink, "@bazel_tools//tools/android:*")'`:

<details>
<summary>Output example</summary>

```
//:tink
//src/main/java/com/google/crypto/tink/signature:signature_pem_keyset_reader
//src/main/java/com/google/crypto/tink:pem_key_type
@bazel_tools//tools/android:desugar_java8_extra_bootclasspath
@bazel_tools//tools/android:java_base_extras.jar
```
</details>

And looking at the [definition for .../tink:pem_key_type](https://github.com/google/tink/blob/2a2c2143c7b951d8c3df12d1cb59fd169531e5a7/java_src/src/main/java/com/google/crypto/tink/BUILD.bazel#L623) we can see that it's an **android_library and not a java_library.**

**Provide your version information:**

-   Language: Java
-   Version: Current Head (`v1.6.0-728-g2a2c2143c`)
-   Environment: Linux

**Is there anything else you’d like to add?**

I unblocked myself by commenting out the offending libraries in a patch:

<details>
  <summary>Patch</summary>

```diff
diff --git a/BUILD.bazel b/BUILD.bazel
index 8418e5016..f6e3dfcde 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -113,9 +113,9 @@ gen_maven_jar_rules(
         "//src/main/java/com/google/crypto/tink/signature:rsa_ssa_pkcs1_verify_key_manager",
         "//src/main/java/com/google/crypto/tink/signature:rsa_ssa_pss_sign_key_manager",
         "//src/main/java/com/google/crypto/tink/signature:rsa_ssa_pss_verify_key_manager",
-        "//src/main/java/com/google/crypto/tink/signature:signature_config",
-        "//src/main/java/com/google/crypto/tink/signature:signature_key_templates",
-        "//src/main/java/com/google/crypto/tink/signature:signature_pem_keyset_reader",
+        # "//src/main/java/com/google/crypto/tink/signature:signature_config",
+        # "//src/main/java/com/google/crypto/tink/signature:signature_key_templates",
+        # "//src/main/java/com/google/crypto/tink/signature:signature_pem_keyset_reader",
         "//src/main/java/com/google/crypto/tink/signature/internal:sig_util",
         "//src/main/java/com/google/crypto/tink/streamingaead:aes_ctr_hmac_streaming_key_manager",
         "//src/main/java/com/google/crypto/tink/streamingaead:aes_gcm_hkdf_streaming_key_manager",
```
</details>

so I believe this problem is self-contained to just those rules, but I assume the real solution is to change `/tink:pem_key_type` to a java_library from an android_library

**Comment by chuckx** [comment]
This should be fixed by 59bb34495d1cb8f9d9dbc0f0a52c4f9e21491a14.

**Comment by notnarb** [comment]
As of that commit it is working for me now, thank you!
